src/pybind/mgr/cephadm/ssh.py

```diff
@@ -189,16 +201,28 @@ async def _write_remote_file(self,
             logger.exception(msg)
             raise OrchestratorError(msg)
 
-    def write_remote_file(self, *args: Any, **kwargs: Any) -> None:
-        self.mgr.event_loop.get_result(self._write_remote_file(*args, **kwargs))
-
-    def _reset_con(self, host: str) -> None:
+    def write_remote_file(self,
+                          host: str,
+                          path: str,
+                          content: bytes,
+                          mode: Optional[int] = None,
+                          uid: Optional[int] = None,
+                          gid: Optional[int] = None,
+                          addr: Optional[str] = None,
+                          ) -> None:
+        self.mgr.event_loop.get_result(self._write_remote_file(
+            host, path, content, mode, uid, gid, addr))
+
+    async def _reset_con(self, host: str) -> None:
         conn = self.cons.get(host)
         if conn:
             logger.debug(f'_reset_con close {host}')
             conn.close()
             del self.cons[host]
 
+    def reset_con(self, host: str) -> None:
+        self.mgr.event_loop.get_result(self._reset_con(host))
+
     def _reset_cons(self) -> None:
```

Comment:
I think you can probably do the same thing to `_reset_cons` and `_reconfig_ssh`. It looks like if `_reset_cons` is called and there are hosts to remove, it will result in the same error as `_reset_con` was causing before